purview/api/v1.py

```diff
@@ -103,4 +103,89 @@ def get_resource_story(resource, uid):
 
     results['meta']['related_nodes'].update(get_corelated_nodes(results_unordered))
 
+    # Adding the artifact itself if it's story is not available
+    if len(results['data']) == 0:
+        results['data'].append(item.serialized_all)
+        results['data'][0]['resource_type'] = item.__label__
+
     return jsonify(results)
+
+
+@api_v1.route('/allstories/<resource>/<uid>')
+def get_resource_all_stories(resource, uid):
+    """
+    Get all unique stories of a resource from Neo4j.
+
+    :param str resource: a resource name that maps to a neomodel class
+    :param str uid: the value of the UniqueIdProperty to query with
+    :return: a Flask JSON response
+    :rtype: flask.Response
+    :raises NotFound: if the item is not found
+    :raises ValidationError: if an invalid resource was requested
+    """
+    item = get_neo4j_node(resource, uid)
+    if not item:
+        raise NotFound('This item does not exist')
+
+    for _, prop_def in item.__all_properties__:
+        if isinstance(prop_def, UniqueIdProperty):
+            forward_query = create_query(item, prop_def.name, uid)
+            backward_query = create_query(item, prop_def.name, uid, reverse=True)
+            break
+
+    if forward_query:
+        results_unordered_forward = get_all_stories(forward_query, [resource])
+    else:
+        results_unordered_forward = []
+
+    if backward_query:
+        results_unordered_backward = get_all_stories(backward_query, [resource])
+    else:
+        results_unordered_backward = []
+
+    all_results = []
+    try:
+        for result_forward in results_unordered_forward:
+            for result_backward in results_unordered_backward:
+                results_unordered = result_forward.copy()
+                results_unordered.update(result_backward)
+                results = OrderedDict({'data': [], 'meta': {}})
+                results['meta']['related_nodes'] = {key: 0 for key in story_flow.keys()}
```

Comment:
You should assign `{key: 0 for key in story_flow.keys()}` to a variable above the for loop. That way you don't recreate this every time.